### DISCUSSION TITLE
Rule to send the supply VCC? - solved

### DISCUSSION THREAD

**Discussion by HoerMirAuf** [arendst/Tasmota#12923] [Q&A] [answered]
Hi.
I use a Battery Powerd ESP12 with Tasmota.
This device will enabled in some cases through the Enable pin.

The only what the device have to do is send a email by sendmail and a rule
All work well 

But ... is there a possibility to send the current Supply VCC also?
Can't find a trigger or command to get the the VCC value.

**Comment by sfromis** [comment, marked as answer]
If you have Tasmota configured for using it (`USE_ADC_VCC`), the VCC detected by the ESP8266 chip is automatically published to MQTT as part of the .../STATE payload, and using JSON-based triggers, a rule can pick it up from there. However, that may not be the best indicator of the battery status, as you normally would have a voltage regulator to give stable 3.3V. Using a voltage divider, you can hook the battery up to the ESP8266 analog input instead.

**Comment by HoerMirAuf** [comment]
Hi.
Yes i know. It's not the best solution to use the Chip VCC detection. But it's good enough for my needs.
HOW can i pick up the Vcc value from a STATE payload? It seem the rule will not consider the STATE and RESULT JSON.

**Reply by barbudor** [reply]
If your ESP12 is powered from the battery through a voltage regulator, measuring chip's VCC is useless because it will always be the value of the regulator's output until it starts to fail.
Best to measure the voltage of the battery
With a LiOn battery, max voltage will be 4.2V bt ESP's ADC pin is limited to [0V, 1.0V] range.
If you connect 2 resitors in a voltage divider like :
```
VBAT ---[R1]---+----ESP_ADC
               +---[R2]--- GND
```
With R1 = 390Kohms and R2 = 100Kohms, the voltage at ESP's ADC will be equal to VBAT * 100/(100+390) ~= 0.2 * VBAT
Then set your ADC input (GPIO17) to ADC Range and use `AdcParams 6` to define a conversion from [0..1024] to [0..4900] and you will directly get the VBAT in the SENSOR message in millivolts.
SENSOR message is directly published to MQTT without any additional rule

If you ever want to publish the value to a different topic than sensor, you can make a rule such as
`ON tele-ANALOG#Range DO publish sometopic %value% ENDON`

**Reply by sfromis** [reply]
If you have a setup with unregulated VCC input (maybe a LiFePO4 battery), these works:
`on Vcc do var1 %value% endon`
`on tele-Vcc do var1 %value% endon`
Obviously, you need a build including `USE_ADC_VCC`

**Comment by HoerMirAuf** [comment]
@barbudor 

No, there is no voltage reglator, the Batterie Power supply direct the ESP Chip.
It's a very simple circuit design but working.

Also there is no MQTT enabled. (But i could if necessary)

To use the ADC Input need a voltage divider but this is a additional load who drain the battery.
I don't need a exact messure ... just a smal voltage indicador

Thank you very much for your help.

@sfromis 
That's just what i'm looking for. Thank you very much.
But .. lol ... it can't be so simple. I tried out every variant like on tele#Vcc or state#Vcc. Damn, it was too simple ^^
Thank you.

EDIT:
Yes, USE_ADC_VCC is included in my tasmota build
